src/platform/plugins/shared/workflows_execution_engine/server/filter_contribution.md

```diff
@@ -0,0 +1,232 @@
+# Adding New Filters/Tags Guide
+
+This guide covers the complete process of adding custom filters or tags to the templating engine.
+
+Adding a new filter/tag requires updates in multiple places:
+
+1. **Engine registration** (server-side)
+2. **UI parsing** (frontend validation)
+3. **Autocompletion** (UI suggestions)
+4. **OSS contribution** (optional, recommended)
+
+## Step 1: Register in the Engine
+
+**File:** [`templating_engine.ts`](./templating_engine.ts)
+
+```typescript
+// src/platform/plugins/shared/workflows_execution_engine/server/templating_engine.ts
+
+constructor() {
+  this.engine = new Liquid({
+    strictFilters: true,
+    strictVariables: false,
+  });
+
+  // Register your custom filter
+  this.engine.registerFilter('my_custom_filter', (value: unknown, ...args: unknown[]): unknown => {
+    // Filter implementation
+    if (typeof value !== 'string') {
+      return value;
+    }
+    // Your filter logic here
+    return transformedValue;
+  });
+}
+```
+
+**Filter Function Signature:**
+```typescript
+(value: unknown, ...args: unknown[]) => unknown
+```
+
+- `value`: The value being filtered (left side of `|`)
+- `args`: Additional filter arguments (right side of `:`)
+- Return: The transformed value
+
+**Example:**
+```typescript
+this.engine.registerFilter('prefix', (value: unknown, prefix: unknown): unknown => {
+  if (typeof value !== 'string' || typeof prefix !== 'string') {
+    return value;
+  }
+  return `${prefix}${value}`;
+});
+
+// Usage: {{ name | prefix: "Mr. " }}
+```
+
+## Step 2: Add to UI Liquid Package
+
+The UI uses liquidjs for validation. Register the filter in the validation instance:
```

Comment:
It's not good, we need to fix this. We should have one place where we register filters.
Let's maybe add TODO here for users to get the idea that it's gonna change?